typescript/schedules.md

```diff
@@ -0,0 +1,1255 @@
+    // delete schedule.spec.intervals;
+
+    // return false to stop retrying early
+  },
+  { retry: retryPolicy }
+);
+
+await schedule.trigger();
+await schedule.backfill({ startAt: new Date(), endAt: addWeeks(new Date(), 1) }); // also takes array
+await schedule.pause('note: pausing');
+await schedule.unpause('now unpause');
+
+await schedule.delete();
+
+const { schedules, nextPageToken } = await client.listByPage({
+  pageSize: 50,
+  nextPageToken: 'base64',
+});
+
+for await (const schedule: Schedule of client.list()) {
+  const { id, memo, searchAttributes } = schedule;
+  // ...
+}
+```
+
+### Types
+
+```ts
+import { DataConverter, LoadedDataConverter } from '@temporalio/common';
+import { loadDataConverter } from '@temporalio/internal-non-workflow-common';
+import {
+  composeInterceptors,
+  Headers,
+  Replace,
+  RetryPolicy,
+  SearchAttributes,
+  Workflow,
+} from '@temporalio/internal-workflow-common';
+import { temporal } from '@temporalio/proto';
+import os from 'os';
+import { Connection } from './connection';
+import { ScheduleClientCallsInterceptor, ScheduleClientInterceptors } from './interceptors';
+import { ConnectionLike, Metadata, WorkflowService } from './types';
+import { WorkflowHandle, WorkflowStartOptions } from './workflow-client';
+
+import type { NonNegativeInteger, RequireAtLeastOne } from 'type-fest';
+
+// TODO is a non-generic NonNegativeInteger possible? The ones below error due to no type argument
+export type PositiveInteger = Exclude<NonNegativeInteger, 0>;
+
+/**
+ * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+ */
+export type Milliseconds = string | NonNegativeInteger;
+
+/**
+ * @format number of seconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+ */
+export type Seconds = string | NonNegativeInteger;
+
+export interface UpdateScheduleOptions {
+  /**
+   * @default TODO
+   */
+  retry?: RetryPolicy;
+}
+
+export interface Backfill {
+  /** Time range to evaluate Schedule in. */
+  startAt: Date;
+  endAt: Date;
+
+  /** Override Overlap Policy for this request. */
+  overlapPolicyOverride?: ScheduleOverlapPolicy;
+}
+
+/**
+ * Handle to a single Schedule
+ */
+export interface ScheduleHandle {
+  /**
+   * This Schedule's identifier
+   */
+  readonly id: string;
+
+  /**
+   * Update the Schedule
+   */
+  update(mutateFn: (schedule: ScheduleDescription) => void | false, options?: UpdateScheduleOptions): Promise<void>;
+
+  /**
+   * Delete the Schedule
+   */
+  delete(): Promise<void>;
+
+  /**
+   * Trigger an Action to be taken immediately
+   */
+  trigger(): Promise<void>;
+
+  /**
+   * Run though the specified time period(s) and take Actions as if that time passed by right now, all at once. The
+   * Overlap Policy can be overridden for the scope of the Backfill.
+   */
+  backfill(options: Backfill | Backfill[]): Promise<void>;
+
+  /**
+   * Pause the Schedule
+   */
+  pause(note?: string): Promise<void>;
```

Comment:
Can you clarify what you do here if already paused?

Also @dnr, can I technically do this in `update` if I wanted? Or does pause have to be done in patch?

Reply:
IE does setting `UpdateScheduleRequest.schedule.state.paused` result in the state changing.

> Can you clarify what you do here if already paused?

I can update comment to say nothing happens. (At least I imagine there's no error?)

Reply:
Nothing happens.

Yes, you can do it in update. The intention is that this is a common case of update, so a dedicated api is easier to use.

Reply:
:+1: I might be inclined to make multiple changes and set paused to true instead of using patch w/ a reason.